### ISSUE TITLE
[BUG] Limited to 16 threads? Missing logfiles?

### ISSUE THREAD

**Issue by xback** [IntelLabs/control-flag#36]
Thank for trying out ControlFlag. We would appreciate if you can fill out the template below for reporting a bug.

**Describe the bug**
Amount of threads seems to be limited to 16 ?  is this expected behaviour?
Also only 4 logfiles are generated, even when more threads are in use. (> 4)

**Exact command to reproduce**
without -j  or using -j20 (Intel i9 10900K. 10c/20t)

**Callstack (if it is a crash bug) or error info**
N/A

**Expected behavior**
> 16 threads and more log files

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Compiler: GCC 9.3.0
 - 32-bit or 64-bit? 64
 - Build command: as documented

**Comment by nhasabni** [comment]
Hi @xback,

The number of scanning threads are limited by default to the number of CPUs on the system. As you found, one can control the number using `-j` option.

The scanning threads are internally divided into several groups (e.g., file handler or logger threads) - this is to ensure that the system is not overwhelmed. That is why the number of log files generated could be less than the number of scanner threads.  So it is an expected behavior.

Thanks.

-- the ControlFlag team

**Comment by xback** [comment]
> The number of scanning threads are limited by default to the number of CPUs on the system.
Thanks for your reply.

But it ran on a 20 logical core cpu?
I'll rerun on our 80-core machine tomorrow.

Regards

**Comment by nhasabni** [comment]
> > The number of scanning threads are limited by default to the number of CPUs on the system.
> > Thanks for your reply.
> 
> But it ran on a 20 logical core cpu? I'll rerun on our 80-core machine tomorrow.
> 
> Regards

Yes, we use `nproc` utility to obtain the number of CPUs on the system, which as per our knowledge, returns the number of logical CPUs.
